### DISCUSSION TITLE
After using SCEE for a while, these 3 "nuisances" I want to figure out the most

### DISCUSSION THREAD

**Discussion by khlsvr** [Helium314/SCEE#544] [Q&A]
I was thinking about posting these or some of them on the issues page, but I would have been required to install SC and test whether the thing I'm explaining also occurs there etc, and I thought I want to keep it less formal so I decided to post them all in one post here in the discussion forum. 

1. SCEE asks on crossings whether there is an island. Here in Finland we have crossings with a table or a hump. I don't know how to get this done without having to resort to edit the tags. Sometimes it adds an extra question, what kind, is it chicane, narrowed road or such etc, I remember some 4 options there. These add traffic_calming key. Would be useful if could choose table or hump there. The custom overlay `i` button takes you to https://github.com/Helium314/SCEE/blob/modified/CONTRIBUTING_A_NEW_QUEST.md#element-selection and I can see there is information how to add new quests but I'm not sure I'm up for that task if this comes to that.

2. I have a consistent problem with the map being zoomed into an old location, not the current location. Also the zoom to current location button is not working properly. 
.
When I switch apps to some other app and then come back to SCEE,  it's at a location where I have submitted edit earlier, or some other location I was earlier. Pressing the button to jump to current location turns the button to green and changes its icon to triange. Pressing again changes it to circle. I have to click some element on map or + button to add something new and then cancel it, then I can press that "current location" button to jump to current location.

3. I would like to be able to see all features at a given location where I am without having to open another app like OSM focus reborn or vespucci. An example case from today: I'm at a railway station overpass. There are elevators to the ground floor. I am answering questions about the surfaces and the stairs. I got interested whether OSM has mapped the elevators there at all. How can I see that quickly in SCEE? I tried to add a custom overlay which would show me any key = any value, but I'm not succeeding. For this particular use case I could see the elevators once I added a custom overlay for nodes that says `highway = elevator` or `highway ~ .+.` or `highway ~ .*.` But what if I wanted to see if there are some nodes where I don't know what the `key` is? The wildcard `.+.` doesn't work for key, so `.+. ~ .+.` shows me nothing.

**Comment by mnalis** [comment]
here is my quick take, but I could be wrong:

> 1. SCEE asks on crossings whether there is an island. Here in Finland we have crossings with a table or a hump

If you enable `POI: Traffic...` quest, when you click on blue circle, you'll have option to say `It's a raised crossing` which will add `traffic_calming=table` on the crossing (it might be present on some crossing quest too? not sure). No such option for `hump` as far as I know, but if you enable `Expert mode`, you'll have `Show/edit tags` on each quest, when you can add any tag you like (they get cached depending on use, so it's not _as horribly clunky_ as it might seem on the first glance)

Oh, and if you do choose to implement a quest in the future, it's not that hard really (you basically copy/paste from similar quest and work from there) and there are people willing to help! But it should follow (more lax than upstream StreetComplete, but still) [SCEE quest guidelines](https://github.com/Helium314/SCEE/blob/modified/QUEST_GUIDELINES.md), so best to first open separate issue to discuss it.

> 2. I have a consistent problem with the map being zoomed into an old location, not the current location

Does it happen only when switching apps? If so, that might be dependent on Android version / manufacturer, as it might lose GPS location when you switch to another app to save power (there was related discussion about that here: https://github.com/streetcomplete/StreetComplete/issues/5581#issuecomment-2068128715). 
- It would help understanding if you could provide video of it happening. 
- Also, running at the same time some other app which keeps **GPS active while running in background** (like e.g. [OsmAnd](https://wiki.openstreetmap.org/wiki/OsmAnd) track recording) might change the outcome, so video of that would be useful too.
- Trying to turn off all power saving for SCEE app might help too (see e.g. https://dontkillmyapp.com/)

> 3. I would like to be able to see all features at a given location where I am without having to open another app like OSM focus reborn or vespucci

Well,  alternatives of what you can do:
- if you want Vespucci/OSMFocus, best is to long-press on the screen and choose `Open location in another app`. It should open at specified location
- you can try `Places` and `Things` overlays for nice UI, but you must know which one you want, and they might not cover everything (they don't cover elevators at the moment for example AFAIK)
- you can enable some of the `POI: ...` SCEE-specific quests. In elevator case, `POI: Traffic...` is the one you want. It will show as small blue circle, and you can click on it (and if you have `expert mode` you can get all tags too). You can also have multiple quest presets, and use quick button to change between them (if you don't want those POIs to be displayed at all times)
- you can create custom overlay by setting filter element  to `!~nonexistent` (which will select all elements which are not tagged with `nonexistent` tag -- which is all of them), but you probably won't like how it looks.

**Comment by Helium314** [comment]
> SCEE asks on crossings whether there is an island. Here in Finland we have crossings with a table or a hump

You can have both, so this is completely orthogonal.
The old crossing type quest had the option of adding a table, but the quest was removed from SC / replaced with some other quest(s?).

> I have a consistent problem with the map being zoomed into an old location, not the current location. Also the zoom to current location button is not working properly.

This is a case for an issue, not a discussion. Would be useful if you checked whether the same thing happens in StreetComplete.

> I would like to be able to see all features at a given location where I am without having to open another app like OSM focus reborn or vespucci

SC(EE) is not made for "showing everything". When you know which specific features you want to see, create a suitable filter and use a custom overlay.
When you _really_ want to see everyting, SCEE is a poor choice (I have an "everything" overlay for testing, and just selecting the feature you want can be horrible).
